src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,378 @@
+    
+    Id = identity_matrix(R,d)
+    Xpre = make_coordinate_matrix_no_identity(d, n, MC, R, x, xdict)
+    return interlace_columns(Id, Xpre, B, R, x)
+end
+
+
+# This function returns all d x d determinants of the matrix X from above
+# of all collections of d-columns coming from the bases of the matroid.
+
+function bases_determinants(X::Matrix{T}, Bs::Vector{Vector{Int}})  where {T<:MPolyElem}
+
+    #d::Int, n::Int, Bs::Vector{Vector{Int}},
+    #MC::Vector{Vector{Int}},
+    #B::Vector{Int}, R::MPolyRing, x::Vector{T},
+    #xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    #X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+    
+    return unique!([det(X[:, b]) for b in Bs ])
+end
+
+
+# This forms the semigroup of the polynomial ring from make_polynomial_ring
+# generated by the determinants from bases_determinants.
+function localizing_semigroup(d::Int, n::Int, Bs::Vector{Vector{Int}},
+                              MC::Vector{Vector{Int}}, B::Vector{Int},
+                              R::MPolyRing, x::Vector{T},
+                              xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    basesX = bases_determinants(d, n, Bs, MC, B, R, x, xdict)
+    
+    sTotal = MPolyPowersOfElement(basesX[1])
+    
+    if length(basesX) == 1
+        return sTotal
+    end
+    
+    for i in 2:length(basesX)
+        if !(basesX[i] in sTotal)
+            sTotal = product(sTotal, MPolyPowersOfElement(basesX[i]))
+        end
+    end
+    
+    return sTotal
+    
+end
+
+
+# This function returns a triple (X, SinvR, Ipre) where X is the matrix
+# from make_coordinate_matrix, SinvR is the localization of the
+# polynomial ring at the semigroup created by localizing_semigroup,
+# and Ipre is the ideal generated by the d x d determinants of nonbases.
+# Note that Ipre is not saturated. 
+function matroid_stratum_matrix_coordinates_given_ring(d::Int, n::Int,
+                                                       M::Matroid,
+                                                       F::AbstractAlgebra.Ring,
+                                                       B::Vector{Int},
+                                                       R::MPolyRing,
+                                                       x::Vector{T},
+                                                       xdict::Dict{Vector{Int}, MPolyElem}) where T <: MPolyElem
+    
+    Bs = bases(M)
+    NBs = nonbases(M)
+    MC = bases_matrix_coordinates(Bs,B)
+
+    
+    X =  make_coordinate_matrix(d, n, MC, B, R, x, xdict)                        
+    basesX = bases_determinants(X, Bs)
+
+    #S = localizing_semigroup(d, n, Bs, MC, B, R, x, xdict)
+    
+    S = MPolyPowersOfElement(R , basesX)
+    SinvR , iota = Localization(R, S)
+    # X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+
+    Igens = unique!([det(X[:, nb]) for nb in NBs ])
+    Iloc = ideal(SinvR, Igens)
+    if iszero(Iloc)
+      return (X, SinvR)
```

Comment:
It seems you deviate from the above comment by now.

Edit: If I remember correctly, this is due to the discussions we once had, right? It seems we didn't adjust the comment.

Reply:
Correct, we didn't adjust the comment